File: Problems/Reverse-Bits/Reverse-Bits.java
public class Solution {
    public int reverseBits(int n) {
        int r = 0;
        for (int b = Integer.SIZE - 1; b >= 0; b--)
            r |= (n >> b & 1) << (Integer.SIZE - 1 - b);
        return r;
    }
}

File: Problems/Reverse-Bits/Reverse-Bits.js
/**
 * @param {number} n - a positive integer
 * @return {number} - a positive integer
 */
var reverseBits = function (n) {
    let r = 0;
    for (let b = 32 - 1; b >= 0; b--)
        r |= (n >>> b & 1) << (32 - 1 - b);
    return r >>> 0;
};
